src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/LegacyPropagator.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    internal class LegacyPropagator : TextMapPropagator
+    {
+        internal static TextMapPropagator Instance { get; } = new LegacyPropagator();
+
+        public override IReadOnlyCollection<string> Fields { get; } = new HashSet<string>() { TraceParent, RequestId, TraceState, Baggage, CorrelationContext };
+
+        public override void Inject(Activity activity, object carrier, PropagatorSetterCallback setter)
+        {
+            if (activity is null || setter is null)
+            {
+                return;
+            }
+
+            string? id = activity.Id;
+            if (id is null)
+            {
+                return;
+            }
+
+            if (activity.IdFormat == ActivityIdFormat.W3C)
+            {
+                setter(carrier, TraceParent, id);
+                if (activity.TraceStateString is not null)
```

Comment:
Should we be testing for string.Empty in addition to null here?

Reply:
It is corner case but I agree if we have space wouldn't make sense to call the setter. I'll change it as your suggested.